contracts/proxy/TransparentUpgradeableProxy.sol

```diff
@@ -131,6 +129,17 @@ contract TransparentUpgradeableProxy is UpgradeableProxy {
         }
     }
 
+    /**
+     * @dev Changes the admin of the proxy.
+     * 
+     * Emits an {AdminChanged} event.
+     */
+    function _changeAdmin(address newAdmin) internal {
+        require(newAdmin != address(0), "TransparentUpgradeableProxy: new admin is the zero address");
```

Comment:
I would have expected the require to be part of the external function, not the internal, eg in case someone wants to implement a `renounceAdmin` function and need to make a `_changeAdmin(0)` call. But if this is in line with the conventions in Contracts, let's go ahead.

Reply:
Okay I took a better look at this and `Owable` only provides the public `transferOwnership` which will check that the sender is the current owner. And `AccessControl` provides `_setupRole` internally _as an exception_ bypassing the role admin because pragmatically it can really useful to set up roles in the constructor.

So if there is a convention it is that this function should not exist internally, and the contract would force this change to go through the admin.

Reply:
To follow this convention I think the right thing to do is to have no internal way to change the admin. I'm comfortable with this, we can reconsider if someone asks for it, but I highly doubt it would happen.